tests/test_fmi3.py

```diff
@@ -1490,6 +1490,57 @@ def test_declared_variable_with_no_type(self):
         with pytest.raises(FMUException, match = re.escape(err)):
             fmu.get_variable_declared_type("String_input")
 
+    def test_set_enum_by_string_name(self):
+        fmu = _get_fmu(FMI3_REF_FMU_PATH / "Feedthrough.fmu", _connect_dll = True)
+        fmu.set_enum(["Enumeration_input"], ["Option 1"])
```

Comment:
`Option 1` might just be the default value? Should add an `assert fmu.get("Enumeration_input")[0] != 1` before that to ensure that `set` actually works.

Same for some more tests below.